src/Transport/Guzzle.php

```diff
@@ -57,13 +58,14 @@ public function exec(Request $request, array $params): Response
 
         $options = [
             'base_uri' => $this->_getBaseUrl($connection),
-            'headers' => [
+            RequestOptions::HEADERS => [
                 'Content-Type' => $request->getContentType(),
             ],
             'exceptions' => false, // 4xx and 5xx is expected and NOT an exceptions in this context
```

Comment:
Will be done for this option too when https://github.com/ruflin/Elastica/pull/1817 will be merged.